tls-client-with-opaque-keys/main.cpp

```diff
@@ -0,0 +1,446 @@
+/*
+ *  Hello world example of a TLS client: fetch an HTTPS page
+ *
+ *  Copyright (C) 2006-2016, Arm Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of Mbed TLS (https://tls.mbed.org)
+ */
+
+/** \file main.cpp
+ *  \brief An example TLS Client application
+ *  This application sends an HTTPS request to os.mbed.com and searches for a string in
+ *  the result.
+ *
+ *  This example is implemented as a logic class (HelloHTTPS) wrapping a TCP socket.
+ *  The logic class handles all events, leaving the main loop to just check if the process
+ *  has finished.
+ */
+
+/* Change to a number between 1 and 4 to debug the TLS connection */
+#define DEBUG_LEVEL 0
+
+#include "mbed.h"
+#include "easy-connect.h"
+
+#include "mbedtls/platform.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/error.h"
+
+#if DEBUG_LEVEL > 0
+#include "mbedtls/debug.h"
+#endif
+
+#include "mbedtls_atca_engine.h"
+
+namespace {
+
+const char *HTTPS_SERVER_NAME = "E112111";
+const int HTTPS_SERVER_PORT = 4433;
+const int RECV_BUFFER_SIZE = 600;
+
+#define GET_REQUEST "GET / HTTP/1.0 \r\n\r\n"
+
+/* personalization string for the drbg */
+const char *DRBG_PERS = "ssl_client";
+
+/* List of trusted root CA certificates
+ * currently only GlobalSign, the CA for os.mbed.com
```

Comment:
Since this sample is meant to be used with a custom server, shouldn't this be removed? A custom server isn't likely to be signed by GlobalSign.